test/src/test/java/hudson/tasks/ArtifactArchiverTest.java

```diff
@@ -248,6 +265,125 @@ public void testDefaultExcludesOff() throws Exception {
         assertTrue(artifacts.child("dir").child(".svn").child("file").exists());
     }
 
+    @Test
+    @Issue("JENKINS-12379")
+    public void testBasePathOK() throws Exception {
+        FreeStyleProject p = j.createFreeStyleProject();
+
+        p.getBuildersList().replaceBy(Collections.singleton(new CreateFilesForBasePathTest()));
+
+        ArtifactArchiver archiver = new ArtifactArchiver("**");
+        archiver.setBasePath("module/dist");
+        p.getPublishersList().replaceBy(Collections.singleton(archiver));
+
+        assertEquals(Result.SUCCESS, build(p));
+        VirtualFile artifacts = p.getBuildByNumber(1).getArtifactManager().root();
+        assertTrue(artifacts.child("target").child("file").exists());
+    }
+
+    @Test
+    @Issue("JENKINS-12379")
+    public void testBasePathUp() throws Exception {
+        FreeStyleProject p = j.createFreeStyleProject();
+
+        p.getBuildersList().replaceBy(Collections.singleton(new CreateFilesForBasePathTest()));
+
+        ArtifactArchiver archiver = new ArtifactArchiver("**");
+        archiver.setBasePath("..");
+        p.getPublishersList().replaceBy(Collections.singleton(archiver));
+
+        assertEquals(Result.FAILURE, build(p));
+        Assert.assertArrayEquals(new VirtualFile[0], p.getBuildByNumber(1).getArtifactManager().root().list());
+    }
+
+    @Test
+    @Issue("JENKINS-12379")
+    public void testBasePathInOutOut() throws Exception {
+        FreeStyleProject p = j.createFreeStyleProject();
+
+        p.getBuildersList().replaceBy(Collections.singleton(new CreateFilesForBasePathTest()));
+
+        ArtifactArchiver archiver = new ArtifactArchiver("**");
+        archiver.setBasePath("foo/../..");
+        p.getPublishersList().replaceBy(Collections.singleton(archiver));
+
+        assertEquals(Result.FAILURE, build(p));
+        Assert.assertArrayEquals(new VirtualFile[0], p.getBuildByNumber(1).getArtifactManager().root().list());
+    }
+
+    @Test
+    @Issue("JENKINS-12379")
+    public void testBasePathAbsolute() throws Exception {
+        FreeStyleProject p = j.createFreeStyleProject();
+
+        p.getBuildersList().replaceBy(Collections.singleton(new CreateFilesForBasePathTest()));
+
+        ArtifactArchiver archiver = new ArtifactArchiver("**");
+        // TODO absolute path on Windows?
```

Comment:
either fix this so it doesn't fail on windows or don't run this test on windows - or remove the TODO

Reply:
still applies